radix-engine/src/model/nodes/bucket.rs

```diff
@@ -293,37 +296,30 @@ impl Bucket {
                 let _: ConsumingBucketBurnInput = scrypto_decode(&args.raw)
                     .map_err(|e| InvokeError::Error(BucketError::InvalidRequestData(e)))?;
 
-                let bucket: Bucket = system_api
-                    .node_drop(&node_id)
-                    .map_err(InvokeError::Downstream)?
-                    .into();
-
-                // Notify resource manager, TODO: Should not need to notify manually
-                let resource_address = bucket.resource_address();
-                let mut value = system_api
-                    .borrow_node_mut(&RENodeId::ResourceManager(resource_address))
+                let node_ref = system_api
+                    .borrow_node(&node_id)
                     .map_err(InvokeError::Downstream)?;
-                let resource_manager = value.resource_manager_mut();
-                resource_manager.burn(bucket.total_amount());
-                if matches!(resource_manager.resource_type(), ResourceType::NonFungible) {
-                    for id in bucket
-                        .total_ids()
-                        .expect("Failed to list non-fungible IDs on non-fungible Bucket")
-                    {
-                        let address = SubstateId(
-                            RENodeId::ResourceManager(resource_address),
-                            SubstateOffset::ResourceManager(ResourceManagerOffset::NonFungible(id)),
-                        );
-                        system_api
-                            .substate_write(
-                                address,
-                                ScryptoValue::from_typed(&NonFungibleSubstate(None)),
-                            )
-                            .map_err(InvokeError::Downstream)?;
-                    }
-                }
+                let bucket = node_ref.bucket();
+                let resource_address = bucket.resource_address();
 
-                Ok(ScryptoValue::from_typed(&()))
+                let bucket_id = match node_id {
+                    RENodeId::Bucket(bucket_id) => bucket_id,
+                    _ => panic!("Unexpected"),
+                };
+
+                system_api
```

Comment:
Can we keep the original TODO comment? I think there may be better ways of signaling updates to total supply over a method call.

Reply:
Yes, sure!